packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -682,9 +714,9 @@ class _RenderCheckbox extends RenderToggleable {
           _drawCheck(canvas, origin, tShrink, strokePaint);
       }
     } else { // Two cases: null to true, true to null
-      final RRect outer = _outerRectAt(origin, 1.0);
+      final Rect outer = _outerRectAt(origin, 1.0);
       final Paint paint = Paint() ..color = _colorAt(1.0);
-      canvas.drawRRect(outer, paint);
+      canvas.drawPath(shape.copyWith(side: side).getOuterPath(outer), paint);
```

Comment:
It would help readability a little if `shape.copyWith(side: side).getOuterPath(outer)` was computed once and bound to a variable with a name like emptyCheckboxPath.